Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -614,16 +614,16 @@ public List<UnspentOutputReference> GetSpendableTransactions(int currentChainHei
                 // When calculating the confirmations the tip must be advanced by one.
 
                 var countFrom = currentChainHeight + 1;
-                var unspentTransactions = address.UnspentTransactions()
-                    .Where(a => countFrom - (a.BlockHeight ?? countFrom) >= confirmations).ToList();
+                var unspentTransactions = address.UnspentTransactions().Select(s =>
+                    new {Transaction = s, Confirmations = s.BlockHeight == null ? 0 : countFrom < s.BlockHeight ? 0 : countFrom - s.BlockHeight }).ToList();
```

Comment:
sorry, but no :)
this `a ? b : c ? d : e` 
i just can't, i don't really understand the change from before here, but it seems that LINQ is used here just to make it unreadable, how about we forget all this LINQ for a moment and go with good old foeach (which we have there anyway) even without those ternary operators, which, if nested, make my brain hurt

```
int countFrom = currentChainHeight + 1;
foreach (TransactionData transactionData in address.UnspentTransactions())
{
    int confirmationCount = 0;
    if (transactionData.BlockHeight != null) 
        confirmationCount = countFrom >= transactionData.BlockHeight ? transactionData.BlockHeight - countFrom : 0;

    if (confirmationCount >= confirmations)
    {
        unspentOutputs.Add(new UnspentOutputReference
        {
            Account = this,
            Address = address,
            Transaction = transactionData
        });
    }
}              
```

unless i've read it wrong, this should be equivalent and i think it is much more readable than LINQ-double-ternary mix

Reply:
I don't think its that hard to understand, though I agree too much linq can be tedious.
But as you went to the trouble of rewriting the method (and it is more readable) I will take your snippet.

Reply:
there were not even brackets, my brain was bleeding ;-)